### PR TITLE
RayCluster with create_settings

### PR DESCRIPTION
**Pull request by al-rigazzi**
In this PR, we change the way Ray is started (internally): instead of creating `SrunSettings`, `AprunSettings`, and `MpirunSettings` objects, we just create a `BaseRunSettings` object with the correct `run_command`. The same holds for batch settings.

Also, notably, this automatically adds support for `Cobalt` as a launcher.

### CODE REVIEW

File: smartsim/experiment.py [hunk 1]
```diff
@@ -207,11 +207,11 @@ def get_status(self, *args):
             statuses = []
             for entity in manifest.models:
                 statuses.append(self._control.get_entity_status(entity))
-            for entity_list in manifest.ensembles:
+            for entity_list in manifest.all_entity_lists:
                 statuses.extend(self._control.get_entity_list_status(entity_list))
-            orchestrator = manifest.db
-            if orchestrator:
-                statuses.extend(self._control.get_entity_list_status(orchestrator))
+            # orchestrator = manifest.db
```
Comment (Spartee) [inline smartsim/experiment.py:212]: meant to be commented out?

File: smartsim/exp/ray/raycluster.py [hunk 2]
```diff
@@ -129,12 +146,23 @@ def __init__(
             interface=interface,
             alloc=alloc,
             num_nodes=num_nodes,
+            run_command=run_command if run_command else "auto",
+            host_list=host_list,
             **kwargs,
         )
         if batch:
-            self._build_batch_settings(num_nodes, time, batch_args, launcher)
+            self.batch_settings = create_batch_settings(
```
Comment (Spartee) [inline smartsim/exp/ray/raycluster.py:154]: Assuming that the previous ``SSUnsupportedError`` catches the cases where ``create_run_settings`` may return an incompatible run command?

File: smartsim/exp/ray/raycluster.py [hunk 3]
```diff
@@ -67,6 +75,10 @@ class RayCluster(EntityList):
     :type batch: bool
     :param time: The walltime the cluster will be running for
     :type time: str
+    :param run_command: specify launch binary. This is currently used only if launcher is Cobalt or PBS.
```
Comment (Spartee) [inline smartsim/exp/ray/raycluster.py:78]: Lets shorten these a bit

**Review by Spartee — CHANGES_REQUESTED** [review]
A few comments. nothing major. Looks good.

File: smartsim/exp/ray/raycluster.py (hunk 2 above)
Comment (al-rigazzi) [reply smartsim/exp/ray/raycluster.py:154]: I cannot think of a case in which this happens and it is not caught. I might be proven wrong, but all test cases passed.

File: smartsim/exp/ray/raycluster.py (hunk 3 above)
Comment (al-rigazzi) [reply smartsim/exp/ray/raycluster.py]: I tried to reduce it to the minimum.

File: smartsim/launcher/lsf/lsfLauncher.py [hunk 4]
```diff
@@ -26,6 +26,8 @@
 
 import time
 
+from smartsim.settings.settings import RunSettings
```
Comment (Spartee) [inline smartsim/launcher/lsf/lsfLauncher.py:29]: use relative improt here

**Review by Spartee — APPROVED** [review]
One minor comment. otherwise LGTM!

**Pull request merged**